crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1178,6 +1178,22 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Removes all components associated with the entity.
+    pub fn clear(&mut self) -> &mut Self {
+        let component_ids: Vec<ComponentId> = self.archetype().components().collect();
```

Comment:
It's unfortunate we have to allocate here. Seems like it should be avoidable since archetypes don't get deleted. But I'm fine with leaving this for now. Especially since archetypes might be deleted in the future where we have relations.